src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java

```diff
@@ -112,12 +115,16 @@ public void consumeMetric(UniversalMetricFrame metric) {
     pointBuilder.tag(Tags.MONITORING_SYSTEM, metric.getMonitoringSystem().toString());
 
     metric.getMetricsList().stream().forEach(val -> {
-      if(!StringUtils.isEmpty(val.getString())) {
-        pointBuilder.addField(val.getName(), val.getString());
-      } else if(val.getFloat()!=0.0) {
-        pointBuilder.addField(val.getName(), val.getFloat());
-      } else if(val.getInt()!=0) {
-        pointBuilder.addField(val.getName(), val.getInt());
+      Metric.ValueCase valueCase = Metric.ValueCase.forNumber(val.getValueCase().getNumber());
+      switch (valueCase) {
```

Comment:
`val.getValueCase()` seems to return a `Metric.ValueCase` so just this should work:
```suggestion
      switch (val.getValueCase()) {
```